### ISSUE TITLE
on alternative backends: adding db.close() to export with limit

### ISSUE THREAD

**Issue by karissa** [maxogden/dat#354]
On the alt-backend branch, adding  `return` here allows us to see the stack trace
 https://github.com/maxogden/dat-core/blob/master/index.js#L975
```
dat $ dat export -d test --limit=1
{"time":"2014-04-30T03:34:57.000Z","latitude":"60.0366","longitude":"-141.2214","depth":"14.6","mag":"1.4","magType":"ml","nst":"","gap":"","dmin":"","rms":"1.51","net":"ak","id":"ak11246293","updated":"2014-04-30T03:39:27.956Z","place":"67km E of Cape Yakataga, Alaska","type":"earthquake"}
Trace: Error: write after end
    at writeAfterEnd (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:144:12)
    at DestroyableTransform.Writable.write (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:192:5)
    at write (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:623:24)
    at flow (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:632:7)
    at DestroyableTransform.pipeOnReadable (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:664:5)
    at DestroyableTransform.emit (events.js:92:17)
    at emitReadable_ (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:448:10)
    at emitReadable (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:444:5)
    at readableAddChunk (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:187:9)
    at DestroyableTransform.Readable.push (/Users/karissa/dev/node_modules/dat-core/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:149:10)
    at done (/Users/karissa/dev/node_modules/dat/bin/export.js:72:15)
    at /Users/karissa/dev/node_modules/dat/node_modules/pump/index.js:72:4
    at f (/Users/karissa/dev/node_modules/dat/node_modules/pump/node_modules/once/once.js:17:25)
    at WriteStream.<anonymous> (/Users/karissa/dev/node_modules/dat/node_modules/pump/index.js:28:19)
    at WriteStream.f (/Users/karissa/dev/node_modules/dat/node_modules/pump/node_modules/once/once.js:17:25)
    at WriteStream.emit (events.js:117:20)
    at WriteStream.stdout.destroy.stdout.destroySoon (node.js:620:16)
    at /Users/karissa/dev/node_modules/dat/node_modules/pump/index.js:42:43
    at call (/Users/karissa/dev/node_modules/dat/node_modules/pump/index.js:49:2)
    at Array.forEach (native)
```

**Comment by maxogden** [comment]
i think that one of the export streams is writing to `process.stdout` when it shouldnt be... cc @mafintosh

**Comment by mafintosh** [comment]
this is a highwatermark bug in the the limitedStream. I'll fix it

**Comment by mafintosh** [comment]
also this commit should be reverted as it might trigger race condition bugs in regards to reading from the db after it has been closed when exporting, https://github.com/maxogden/dat/commit/11286282504e731aaf52c4bf015f370d527b1e86

**Comment by mafintosh** [comment]
the hwm bug is fixed in dat-core 3.8.4

**Comment by karissa** [comment]
@mafintosh got it, thanks!
